### PR TITLE
Changed front end linter

### PR DESCRIPTION
**Pull request by mjvkeulen**
## Description
Reworked front end to adhere to a more strict linter combination.

### CODE REVIEW

File: src/vue/src/components/journal/CommentCard.vue [hunk 1]
```diff
@@ -20,62 +27,86 @@
...
+                            class="timestamp">
+                            {{ $root.beautifyDate(comment.creation_date) }}<br>
                         </span>
                         <span
-                            v-else-if="comment.published"
-                            class="timestamp"
                             v-b-tooltip
-                            :title="'Last edit by: ' + comment.last_edited_by">
+                            v-else-if="comment.published"
+                            :title="'Last edit by: ' + comment.last_edited_by"
```
Comment (Larspolo) [inline src/vue/src/components/journal/CommentCard.vue]: \`\`

File: src/vue/src/components/journal/JournalNonStudent.vue [hunk 2]
```diff
@@ -160,62 +289,53 @@ export default {
...
-            }
-
-            if (!this.discardChanges()) {
-                return
-            }
-
-            if (this.currentNode === -1 || this.currentNode === this.nodes.length ||
-                this.nodes[this.currentNode].type !== 'e' ||
-                this.nodes[this.currentNode].type !== 'd') {
+                /* TODO fix mess */
```
Comment (Larspolo) [inline src/vue/src/components/journal/JournalNonStudent.vue]: Missing `                return this.currentNode`. Also, the "TODO" is still here, did you want to do this in this PR, or later on?

File: src/vue/src/components/journal/JournalNonStudent.vue [hunk 3]
```diff
@@ -229,11 +349,13 @@ export default {
...
-                if ((this.$refs['entry-template-card'].grade !== this.nodes[this.currentNode].entry.grade ||
-                    this.$refs['entry-template-card'].published !== this.nodes[this.currentNode].entry.published) &&
-                    !confirm('Progress will not be saved if you leave. Do you wish to continue?')) {
+            if (this.currentNode !== -1
+                && this.currentNode !== this.nodes.length
+                && (this.nodes[this.currentNode].type === 'e'
+                || (this.nodes[this.currentNode].type === 'd' && this.nodes[this.currentNode].entry !== null))) {
+                if ((this.$refs['entry-template-card'].grade !== this.nodes[this.currentNode].entry.grade
+                        || this.$refs['entry-template-card'].published !== this.nodes[this.currentNode].entry.published)
+                        && !window.confirm('Progress will not be saved if you leave. Do you wish to continue?')) {
```
Comment (Larspolo) [inline src/vue/src/components/journal/JournalNonStudent.vue]: This now uses two different styles of indentation. One is indented by 3 the other by 8

File: src/vue/src/components/lti/LinkAssignment.vue [hunk 4]
```diff
@@ -1,61 +1,75 @@
...
             assignmentAPI.list(this.page.cID)
-                .then(assignments => { this.assignments = assignments })
+                .then((assignments) => { this.assignments = assignments })
         },
         linkAssignment (aID, aLtiCouples) {
-            if (!aLtiCouples || confirm('This assignment is already linked to ' + (aLtiCouples > 1 ? aLtiCouples + ' ' : 'an') + 'other assignment' + (aLtiCouples > 1 ? 's' : '') + ' from the learning-environment, are you sure you also want to link it?')) {
-                assignmentAPI.update(aID, {lti_id: this.lti.ltiAssignID,
+            if (!aLtiCouples || window.confirm(
+                `This assignment is already linked to ${aLtiCouples > 1 ? `${aLtiCouples} ` : 'an'}
+                other assignment${aLtiCouples > 1 ? 's' : ''} from the learning-environment, are you sure you also
```
Comment (Larspolo) [inline src/vue/src/components/lti/LinkAssignment.vue]: maybe change it to: "are you sure you want to link this one as well?"

File: src/vue/src/components/lti/LinkCourse.vue [hunk 5]
```diff
@@ -1,41 +1,45 @@
...
+        <div
+            v-for="c in courses"
+            :key="c.id">
             <div v-if="c.lti_couples">
                 <main-card
-                    class="orange-border"
-                    @click.native="linkCourse(c)"
                     :line1="c.name"
-                    :line2="c.startdate.substring(0, 4) + '-' + c.enddate.substring(0, 4)"/>
+                    :line2="c.startdate.substring(0, 4) + '-' + c.enddate.substring(0, 4)"
```
Comment (Larspolo) [inline src/vue/src/components/lti/LinkCourse.vue]: this can also use some \`\`

File: src/vue/src/components/lti/LinkCourse.vue [hunk 6]
```diff
@@ -1,41 +1,45 @@
...
+                    class="orange-border"
+                    @click.native="linkCourse(c)"/>
             </div>
             <div v-else>
                 <main-card
-                    class="green-border"
-                    @click.native="linkCourse(c)"
                     :line1="c.name"
-                    :line2="c.startdate.substring(0, 4) + '-' + c.enddate.substring(0, 4)"/>
+                    :line2="c.startdate.substring(0, 4) + '-' + c.enddate.substring(0, 4)"
```
Comment (Larspolo) [inline src/vue/src/components/lti/LinkCourse.vue]: and these

File: src/vue/src/components/template/TemplateEdit.vue [hunk 7]
```diff
@@ -2,62 +2,106 @@
...
                 Preview
             </b-button>
         </div>
-        <hr/>
-        <div v-if="this.mode == 'edit'">
-            <b-input class="mr-sm-2 multi-form theme-input" id="template-name" v-model="template.name" placeholder="Template name" required/>
-            <div v-if="!formatSettings.available" class="template-availability">
-                <b-button v-on:click.stop @click="toggleActive" class="delete-button">
+        <hr>
+        <div v-if="mode == 'edit'">
```
Comment (Larspolo) [inline src/vue/src/components/template/TemplateEdit.vue]: ===

File: src/vue/src/components/template/TemplateEdit.vue [hunk 8]
```diff
@@ -88,22 +135,37 @@
...
+                            <icon
+                                class="trash-icon"
+                                name="trash"
+                                scale="1.75"
+                                @click.native="removeField(field.location)"/>
                         </b-col>
                     </b-row>
                 </b-card>
-                <div class="invisible"></div>
+                <div class="invisible"/>
```
Comment (Larspolo) [inline src/vue/src/components/template/TemplateEdit.vue]: So why do we close this with an extra `/`, but for `hr` tags you removed them?

File: src/vue/src/components/template/TemplatePreview.vue [hunk 9]
```diff
@@ -68,28 +76,26 @@
...
+        sandboxedIframe,
     },
+    props: ['template'],
     methods: {
         parseSelectionOptions (fieldOptions) {
             if (!fieldOptions) {
                 return [{ value: null, text: 'Please select an option...' }]
             }
-            var options = JSON.parse(fieldOptions).filter(e => e).map(x => { return { value: x, text: x } })
+            const options = JSON.parse(fieldOptions).filter(e => e).map(x => ({ value: x, text: x }))
```
Comment (Larspolo) [inline src/vue/src/components/template/TemplatePreview.vue]: I assume it doesnt work without the `()`? as that will act like a function? If so, maybe change it to `Object({ value: x, text: x })` to make it a bit more clear imo

File: src/vue/src/main.js [hunk 10]
```diff
@@ -1,33 +1,37 @@
 import Vue from 'vue'
-import App from './App'
-import router from './router'
-import store from './store'
 import axios from 'axios'
 import BootstrapVue from 'bootstrap-vue'
 import 'bootstrap/dist/css/bootstrap.css'
 import 'bootstrap-vue/dist/bootstrap-vue.css'
-import 'flatpickr/dist/flatpickr.css'
+import 'flatpickr/dist/flatpickr.css' // eslint-disable-line import/no-extraneous-dependencies
```
Comment (Larspolo) [inline src/vue/src/main.js]: ?

File: src/vue/src/router/index.js [hunk 11]
```diff
@@ -2,116 +2,120 @@ import Vue from 'vue'
...
     }, {
         path: '*',
         name: 'NotFound',
         component: ErrorPage,
-        props: { code: '404', reasonPhrase: 'Not Found', description: `We're sorry but we can't find the page you tried to access.` }
-    }]
+        props: {
+            code: '404',
+            reasonPhrase: 'Not Found',
+            description: 'We\'re sorry but we can\'t find the page you tried to access.',
```
Comment (Larspolo) [inline src/vue/src/router/index.js]: I preferred the old method of using \`\` here. Is that not allowed anymore?

File: src/vue/src/store/modules/permissions.js [hunk 12]
```diff
@@ -2,34 +2,47 @@ import * as types from '../constants/permission-types.js'
...
             return permissions[givenKeyLevel][permission]
         } else if ('cID' in routeParams && types.COURSE_LEVEL_PERMISSIONS.has(permission)) {
-            return permissions['course' + routeParams.cID][permission]
+            return permissions[`course${routeParams.cID}`][permission]
         } else if ('aID' in routeParams && types.ASSIGNMENT_LEVEL_PERMISSIONS.has(permission)) {
-            return permissions['assignment' + routeParams.aID][permission]
+            return permissions[`assignment${routeParams.aID}`][permission]
         } else if (types.GENERAL_LEVEL_PERMISSIONS.has(permission)) {
-            return permissions['general'][permission]
+            return permissions.general[permission]
```
Comment (Larspolo) [inline src/vue/src/store/modules/permissions.js]: Now this is inconsistant with the above

File: src/vue/src/views/Assignment.vue [hunk 13]
```diff
@@ -242,105 +369,20 @@ export default {
...
-            var points = 0
+            let needsMarking = 0
+            let unpublished = 0
+            let points = 0
 
-            for (var i = 0; i < filteredJournals.length; i++) {
-                needsMarking += filteredJournals[i]['stats']['submitted'] - filteredJournals[i]['stats']['graded']
-                unpublished += filteredJournals[i]['stats']['submitted'] - filteredJournals[i]['stats']['published']
-                points += filteredJournals[i]['stats']['acquired_points']
+            for (let i = 0; i < filteredJournals.length; i++) {
```
Comment (Larspolo) [inline src/vue/src/views/Assignment.vue]: can be a foreach

File: src/vue/src/views/AssignmentsOverview.vue [hunk 14]
```diff
@@ -1,31 +1,61 @@
...
+                    :deadline="d"
+                    :course="d.course" />
             </b-link>
-            <b-link v-else v-for="(course, j) in d.courses" tag="b-button" :to="assignmentRoute(course.id, d.id, d.journal, d.is_published)"  :key="i + '-' + j">
-                <todo-card :deadline="d" :course="course" />
+            <b-link
+                v-for="(course, j) in d.courses"
+                v-else
+                :to="assignmentRoute(course.id, d.id, d.journal, d.is_published)"
+                :key="i + '-' + j"
```
Comment (Larspolo) [inline src/vue/src/views/AssignmentsOverview.vue]: \`\`

File: src/vue/src/views/AssignmentsOverview.vue [hunk 15]
```diff
@@ -142,7 +138,41 @@ export default {
...
+                route.name = 'FormatEdit'
+            } else if (this.$hasPermission('can_view_all_journals', 'assignment', aID)) {
+                route.name = 'Assignment'
+            } else {
+                route.name = 'Journal'
+                route.params.jID = jID
+            }
+            return route
+        },
+        compare (a, b) {
```
Comment (Larspolo) [inline src/vue/src/views/AssignmentsOverview.vue]: These kind of functions should be extracted one day as well imo

File: src/vue/src/views/Home.vue [hunk 16]
```diff
@@ -1,16 +1,24 @@
...
+        <div
+            v-for="c in courses"
+            slot="main-content-column"
+            :key="c.id">
             <b-link :to="{ name: 'Course', params: { cID: c.id, courseName: c.name } }">
                 <main-card
                     :line1="c.name"
-                    :line2="c.startdate ? (c.startdate.substring(0, 4) + (c.enddate ? ' - ' + c.enddate.substring(0, 4) : '')) : ''"
+                    :line2="c.startdate ? (c.startdate.substring(0, 4) +
+                    (c.enddate ? ' - ' + c.enddate.substring(0, 4) : '')) : ''"
```
Comment (Larspolo) [inline src/vue/src/views/Home.vue]: missing \`\`

File: src/vue/src/views/LtiLaunch.vue [hunk 17]
```diff
@@ -71,23 +82,57 @@ export default {
...
+        this.ltiJWT = this.$route.query.ltiJWT
+
+        this.loadLtiData().then(() => {
+            this.handleInitialState()
+            /* The lti parameters such as course id are not set if we are a student. */
+            if (this.lti.ltiCourseID) {
+                courseAPI.getLinkable().then((courses) => {
+                    if (courses.length) {
+                        this.courses = courses
+                        this.states.state = this.tempStateToCheckIfWeCanAutoSetup
```
Comment (Larspolo) [inline src/vue/src/views/LtiLaunch.vue]: `tempStateToCheckIfWeCanAutoSetup` => `checkCanAutoSetup`?

File: src/vue/src/views/UserRoleConfiguration.vue [hunk 18]
```diff
@@ -6,8 +6,13 @@
                 <thead>
                     <tr>
                         <th/>
-                        <th v-for="role in roles" :key="'th-' + role">
-                            <b-button v-if="!undeleteableRoles.includes(role)" @click="deleteRole(role)" class="delete-button">
+                        <th
+                            v-for="role in roles"
+                            :key="'th-' + role">
```
Comment (Larspolo) [inline src/vue/src/views/UserRoleConfiguration.vue]: \`\`, but why even add that prefix?

File: src/vue/src/views/UserRoleConfiguration.vue [hunk 19]
```diff
@@ -16,17 +21,23 @@
...
                 <tbody>
-                    <tr v-for="permission in permissions" :key="permission">
+                    <tr
+                        v-for="permission in permissions"
+                        :key="permission">
                         <td class="permission-column"><b>{{ formatPermissionString(permission) }}</b></td>
-                        <td v-for="role in roles" :key="role + '-' + permission">
+                        <td
+                            v-for="role in roles"
+                            :key="role + '-' + permission">
```
Comment (Larspolo) [inline src/vue/src/views/UserRoleConfiguration.vue]: \`\`

File: src/vue/src/api/connection.js [hunk 20]
```diff
@@ -3,23 +3,23 @@ import axios from 'axios'
 export default {
     conn: axios.create({
         baseURL: 'http://localhost:8000/',
-        transformRequest: [(data) => JSON.stringify(data)],
+        transformRequest: [data => JSON.stringify(data)],
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/api/connection.js]: Should be enforced, not sure if it's dependant on following a statement (return) or block. I will look into it.

File: src/vue/src/api/connection.js [hunk 21]
```diff
@@ -3,23 +3,23 @@ import axios from 'axios'
...
         headers: {
-            'Accept': 'application/json',
-            'Content-Type': 'application/json'
-        }
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/api/connection.js]: Quotes cannot be removed with keys composed of a dash. Personally I feel these simply should be the exception, and only then.

File: src/vue/src/components/assets/DeadlineDeck.vue [hunk 22]
```diff
@@ -1,17 +1,35 @@
...
+                    :deadline="d"
+                    :course="d.course" />
             </b-link>
-            <b-link v-else v-for="(course, j) in d.courses" tag="b-button" :to="assignmentRoute(course.id, d.id, d.journal, d.is_published)"  :key="i + '-' + j">
-                <todo-card :deadline="d" :course="course" />
+            <b-link
+                v-for="(course, j) in d.courses"
+                v-else
+                :to="assignmentRoute(course.id, d.id, d.journal, d.is_published)"
+                :key="i + '-' + j"
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assets/DeadlineDeck.vue]: I need to look at why the rule is not hitting these hm

File: src/vue/src/components/journal/JournalNonStudent.vue (hunk 2 above)
Comment (mjvkeulen) [reply src/vue/src/components/journal/JournalNonStudent.vue]: If I recall correctly the return value is never used and this function was just a mess hence the TODO. Not sure could do it here, but I do want to get through this rather swiftly as conflicts stack so quickly.

File: src/vue/src/components/assets/Feedback.vue [hunk 23]
```diff
@@ -95,21 +92,22 @@ export default {
...
-                var data = new FormData()
+                const data = new FormData()
                 data.append('topic', this.topic)
                 data.append('ftype', this.type)
                 data.append('feedback', this.feedback)
                 data.append('user_agent', navigator.userAgent)
                 data.append('url', window.location.href)
                 if (this.files != null) {
-                    for (var i = 0; i < this.files.length; i++) {
+                    for (let i = 0; i < this.files.length; i++) {
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assets/Feedback.vue]: Prob many loops can be changed, I only tackled iterators. (`no-iterator` and `no-restricted-syntax`)

File: src/vue/src/components/template/TemplateEdit.vue (hunk 8 above)
Comment (mjvkeulen) [reply src/vue/src/components/template/TemplateEdit.vue]: https://stackoverflow.com/questions/1946426/html-5-is-it-br-br-or-br second answer, not quite sure what the rule was again.

File: src/vue/src/components/template/TemplatePreview.vue (hunk 9 above)
Comment (mjvkeulen) [reply src/vue/src/components/template/TemplatePreview.vue]: I think it might work without the `()` and it would be more clear, otherwise I do prefer the specificity of `Object(..)`

File: src/vue/src/main.js (hunk 10 above)
Comment (mjvkeulen) [reply src/vue/src/main.js]: All imports need to resolve, if global imports are used they need to be listed in our dependencies. Not sure if I remember why I couldn't just add them to our dependencies however hm

File: src/vue/src/components/assets/DeadlineDeck.vue (hunk 22 above)
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assets/DeadlineDeck.vue]: `prefer-template` and `no-useless-concat` are not applied in javascript contexts of templates. I do think they should be but fixing the rules hm

File: src/vue/src/components/journal/JournalNonStudent.vue (hunk 3 above)
Comment (mjvkeulen) [reply src/vue/src/components/journal/JournalNonStudent.vue]: Quite a bit of freedom is currently allowed, do you prefer:
```
            if (this.currentNode !== -1
            && this.currentNode !== this.nodes.length
            && (this.nodes[this.currentNode].type === 'e'
            || (this.nodes[this.currentNode].type === 'd' && this.nodes[this.currentNode].entry !== null))) {
                if ((this.$refs['entry-template-card'].grade !== this.nodes[this.currentNode].entry.grade
                || this.$refs['entry-template-card'].published !== this.nodes[this.currentNode].entry.published)
                && !window.confirm('Progress will not be saved if you leave. Do you wish to continue?')) {
                    return false
                }
            }
```
or

```
            if (this.currentNode !== -1
                && this.currentNode !== this.nodes.length
                && (this.nodes[this.currentNode].type === 'e'
                || (this.nodes[this.currentNode].type === 'd' && this.nodes[this.currentNode].entry !== null))) {
                if ((this.$refs['entry-template-card'].grade !== this.nodes[this.currentNode].entry.grade
                    || this.$refs['entry-template-card'].published !== this.nodes[this.currentNode].entry.published)
                    && !window.confirm('Progress will not be saved if you leave. Do you wish to continue?')) {
                    return false
                }
            }
```

Comment (Larspolo) [reply src/vue/src/components/journal/JournalNonStudent.vue]: second, I prefer spaces for inside the if-stmt also

File: src/vue/src/api/connection.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/api/connection.js]: Ah right, for arrow functions with a single statement (no block and thus return) and only a single argument this is the preferred style. 

I think that's clean and used often by us e.g. `.then(response => response.data.assignment)`

File: src/vue/src/components/journal/JournalNonStudent.vue (hunk 3 above)
Comment (mjvkeulen) [reply src/vue/src/components/journal/JournalNonStudent.vue]: spaces for inside the if-stmt?

File: src/vue/src/components/template/TemplatePreview.vue (hunk 9 above)
Comment (Larspolo) [reply src/vue/src/components/template/TemplatePreview.vue]: Seems that it wouldn't work without the `()`
image

File: src/vue/src/components/course/CourseMembersEditor.vue [hunk 24]
```diff
@@ -1,73 +1,135 @@
...
-      :roles="roles" />
+    <div>
+        <h4 class="mb-2"><span>Manage course members</span></h4>
+        <div class="d-flex">
+            <input
+                v-if="viewEnrolled"
+                v-model="searchValue"
+                class="theme-input flex-grow-1 no-width multi-form mr-2"
+                type="text"
+                placeholder="Search..." >
```
Comment (engelhamer) [inline src/vue/src/components/course/CourseMembersEditor.vue]: Again some inconsistency: this time there's a space before `>` but this is not always the case.

File: src/vue/src/components/entry/EntryFields.vue [hunk 25]
```diff
@@ -118,43 +147,49 @@ import pdfDisplay from '@/components/assets/PdfDisplay.vue'
 import sandboxedIframe from '@/components/assets/SandboxedIframe.vue'
 
 export default {
+    components: {
+        'file-upload-input': fileUploadInput,
```
Comment (engelhamer) [inline src/vue/src/components/entry/EntryFields.vue]: Again a lot of excessive explicit kebab definitions.

File: src/vue/src/components/entry/EntryNonStudentPreview.vue [hunk 26]
```diff
@@ -35,45 +57,57 @@
...
 <script>
 import commentCard from '@/components/journal/CommentCard.vue'
 import entryFields from '@/components/entry/EntryFields.vue'
-import entryAPI from '@/api/entry'
-import icon from 'vue-awesome/components/Icon'
+import entryAPI from '@/api/entry.js'
 
 export default {
+    components: {
+        'comment-card': commentCard,
```
Comment (engelhamer) [inline src/vue/src/components/entry/EntryNonStudentPreview.vue]: Kebabs.

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 27]
```diff
@@ -16,30 +20,34 @@
...
-                        <icon name="calendar" scale="1.8"/>
+                        :class="{'selected': currentPreset.type === 'd'}"
+                        class="change-button preset-type-button float-left mr-3 mt-2 no-hover">
+                        <icon
+                            name="calendar"
+                            scale="1.8"/>
                     </b-button>
                     <div>
-                        <b>Entry</b><br/>
+                        <b>Entry</b><br>
```
Comment (engelhamer) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: Break tags should always be closed.

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 28]
```diff
@@ -52,50 +60,68 @@
...
             </h2>
-            <b-form-select v-model="currentPreset.template" class="multi-form">
-                <option disabled value="">Please select a template</option>
-                <option v-for="template in templates" :key="template.t.tID" :value="template.t">
+            <b-form-select
+                v-model="currentPreset.template"
+                class="multi-form">
+                <option
+                    disabled
+                    value="">Please select a template</option>
```
Comment (engelhamer) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: Option closed on same line whereas this is not done below.

File: src/vue/src/components/journal/CommentCard.vue [hunk 29]
```diff
@@ -5,12 +5,19 @@
...
+                <img
+                    :src="comment.author.profile_picture"
+                    class="profile-picture-sm no-hover">
+                <b-card
+                    :class="$root.getBorderClass($route.params.cID)"
+                    class="no-hover comment-card">
                     <div v-if="!editCommentStatus[index]">
                         <sandboxed-iframe :content="comment.text"/>
-                        <hr class="full-width"/>
+                        <hr class="full-width">
```
Comment (engelhamer) [inline src/vue/src/components/journal/CommentCard.vue]: This should also be closed. Why did you change this?

File: src/vue/src/components/journal/JournalEndCard.vue [hunk 30]
```diff
@@ -1,18 +1,20 @@
...
+    <b-card
+        :class="$root.getBorderClass($route.params.cID)"
+        class="no-hover">
         <h2>End of assignment</h2>
-        <hr class="full-width"/>
+        <hr class="full-width">
         <b v-if="assignment.due_date">
             <span v-if="new Date() > new Date(assignment.due_date) && !assignment.lock_date">
-                The due date for this assignment has passed.<br/>
+                The due date for this assignment has passed.<br>
```
Comment (engelhamer) [inline src/vue/src/components/journal/JournalEndCard.vue]: Again lots of non-closed breaks and lines here.

File: src/vue/src/components/journal/JournalNonStudent.vue [hunk 31]
```diff
@@ -89,13 +141,25 @@ import journalStartCard from '@/components/journal/JournalStartCard.vue'
...
 import store from '@/Store.vue'
-import journalAPI from '@/api/journal'
-import assignmentAPI from '@/api/assignment'
+import journalAPI from '@/api/journal.js'
+import assignmentAPI from '@/api/assignment.js'
 import { mapGetters, mapMutations } from 'vuex'
 
 export default {
+    components: {
+        'entry-non-student-preview': entryNonStudentPreview,
```
Comment (engelhamer) [inline src/vue/src/components/journal/JournalNonStudent.vue]: Large list of useless kebab.

File: src/vue/src/components/lti/LinkAssignment.vue (hunk 4 above)
Comment (engelhamer) [reply src/vue/src/components/lti/LinkAssignment.vue]: And learning-environment -> LMS.

File: src/vue/src/components/profile/NotificationCard.vue [hunk 32]
```diff
@@ -11,35 +13,54 @@
...
 <script>
 import Switch from '@/components/assets/SwitchComponent.vue'
-import preferencesAPI from '@/api/preferences'
+import preferencesAPI from '@/api/preferences.js'
 
 export default {
-    props: ['userData'],
     components: {
-        'toggle-switch': Switch
+        'toggle-switch': Switch,
```
Comment (engelhamer) [inline src/vue/src/components/profile/NotificationCard.vue]: Also a good candidate for a component rename.

File: src/vue/src/views/Assignment.vue [hunk 33]
```diff
@@ -116,44 +159,127 @@ import breadCrumb from '@/components/assets/BreadCrumb.vue'
...
-import icon from 'vue-awesome/components/Icon'
+import assignmentAPI from '@/api/assignment.js'
+import groupAPI from '@/api/group.js'
+import participationAPI from '@/api/participation.js'
 import { mapGetters, mapMutations } from 'vuex'
 
 export default {
     name: 'Assignment',
+    components: {
+        'content-columns': contentColumns,
```
Comment (engelhamer) [inline src/vue/src/views/Assignment.vue]: Kebabs.

File: src/vue/src/components/assets/ProgressBar.vue [hunk 34]
```diff
@@ -3,52 +3,62 @@
         <h5 class="progress-percentage">{{ progressPercentage }}%</h5>
         <h5>
             <b>{{ currentPoints ? currentPoints : 0 }}</b> Points
-            <tooltip v-if="bonusPoints != 0" :tip="(currentPoints - bonusPoints) + ' journal points + ' + bonusPoints + ' bonus points'" />
+            <tooltip
+                v-if="bonusPoints != 0"
+                :tip="`${currentPoints - bonusPoints} journal points ${bonusPoints} bonus points`"/>
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assets/ProgressBar.vue]: Good catch

File: src/vue/src/components/journal/JournalEndCard.vue (hunk 30 above)
Comment (mjvkeulen) [reply src/vue/src/components/journal/JournalEndCard.vue]: I went over this with lars, should be in one of the resolved convos

Comment (mjvkeulen) [reply src/vue/src/components/journal/JournalEndCard.vue]: It boils down to these (`<br>`, `<hr>` etc) being void elements and both manners are legal. However, as they can never hold any content what are we closing? Then again as both are legal, we can agree on one style or the other.

File: src/vue/.eslintrc.js [hunk 35]
```diff
@@ -1,29 +1,47 @@
...
+    rules: {
+        'generator-star-spacing': 'off',
+        'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
+        'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
+        'semi': 0, // do not require ;
+        'indent': ['error', 4],
+        'vue/html-indent': ['error', 4],
+        'vue/require-prop-types': 'off', // Do not demand component property types
+        'vue/script-indent': 'off', // clash with base eslint for some edge cases
+        'vue/attribute-hyphenation': ['error', 'never'], // allow component properties as camelCase
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/.eslintrc.js]: Yes there is one: `vue/component-name-in-template-casing`

For a list of vue specific ones see: 
https://github.com/vuejs/eslint-plugin-vue/tree/master/docs/rules

**Comment by mjvkeulen** [comment]
> Almost done reading: only some views left. Couple of comments:
> 
> * There appears to be some inconsistency with closing forward slashes (with a space in front or not)
> * There is a lot of unnecessary explicit kebab-case component renaming going on
> * Are we not closing breaks and horizontal lines anymore? I believe there are a few mixed cases

The spacing issue could be addressed by:
https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/html-closing-bracket-spacing.md

We would prefer immediate closing right? E.g. `</div>`

On the topic of which line the closing bracket should be placed there is the following rule:
https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/html-closing-bracket-newline.md

I am still in favor of placing it on a new line, did we end up voting against this?

File: src/vue/src/main.js (hunk 10 above)
Comment (mjvkeulen) [reply src/vue/src/main.js]: So `vue-flatpickr-component` is listed in the dependencies, but this import looks for `flatpickr`. I think ignoring this case is fine?

File: src/vue/src/components/profile/NotificationCard.vue (hunk 32 above)
Comment (mjvkeulen) [reply src/vue/src/components/profile/NotificationCard.vue]: `switch` is reserved right?

Comment (engelhamer) [reply src/vue/src/components/profile/NotificationCard.vue]: Yeah, so call the component `ToggleSwitch.vue` so we don't have to rename here.
